muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/guider/GuiderRESTService.java

```diff
@@ -1165,4 +1207,178 @@ else if (value instanceof Double) {
+    }
+    
+    UserEntity studentEntity = userEntityController.findUserEntityById(userEntityId);
+    SchoolDataIdentifier studentIdentifier = studentEntity.defaultSchoolDataIdentifier();
+
+    if (!userSchoolDataController.amICounselor(studentIdentifier) && (!loggedUserRole.equals(EnvironmentRoleArchetype.ADMINISTRATOR) && !loggedUserRole.equals(EnvironmentRoleArchetype.MANAGER) && !loggedUserRole.equals(EnvironmentRoleArchetype.STUDY_PROGRAMME_LEADER))) {
+      return Response.status(Status.FORBIDDEN).build();
+    }
+
+    User student = userController.findUserByDataSourceAndIdentifier(studentIdentifier.getDataSource(), studentIdentifier.getIdentifier());
+    
+    if (student == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    Workspace workspace = workspaceController.findWorkspace(workspaceEntity);
+
+    SchoolDataIdentifier workspaceIdentifier = new SchoolDataIdentifier(workspace.getIdentifier(), workspace.getSchoolDataSource());
+
+    WorkspaceUserEntity workspaceUserEntity = workspaceUserEntityController.findWorkspaceUserEntityByWorkspaceAndUserIdentifierIncludeArchived(workspaceEntity, studentIdentifier);
+    if (workspaceUserEntity != null && workspaceUserEntity.getArchived() == Boolean.TRUE) {
+      workspaceUserEntityController.unarchiveWorkspaceUserEntity(workspaceUserEntity);
+    }
+    if (workspaceUserEntity != null && workspaceUserEntity.getActive() == Boolean.FALSE) {
+      workspaceUserEntityController.updateActive(workspaceUserEntity, Boolean.TRUE);
+      userIndexer.indexUser(workspaceUserEntity.getUserSchoolDataIdentifier().getUserEntity());
+    }
+
+    fi.otavanopisto.muikku.schooldata.entity.WorkspaceUser workspaceUser = workspaceController.findWorkspaceUserByWorkspaceAndUser(workspaceIdentifier, studentIdentifier);
+    if (workspaceUser == null) {
+      workspaceUser = workspaceController.createWorkspaceUser(workspace, student, WorkspaceRoleArchetype.STUDENT);
+      waitForWorkspaceUserEntity(workspaceEntity, studentIdentifier);
+    }
+    else {
+      workspaceController.updateWorkspaceStudentActivity(workspaceUser, true);
+    }
+
+    List<WorkspaceUserEntity> workspaceTeachers = workspaceUserEntityController.listActiveWorkspaceStaffMembers(workspaceEntity);
+    List<UserEntity> recipients = new ArrayList<UserEntity>();
+
+    String workspaceName = workspace.getName();
+    if (!StringUtils.isBlank(workspace.getNameExtension())) {
+      workspaceName += String.format(" (%s)", workspace.getNameExtension()); 
+    }
+
+    String userName = student.getNickName() == null
+      ? student.getDisplayName()
+      : String.format("%s \"%s\" %s (%s)", student.getFirstName(), student.getNickName(), student.getLastName(), student.getStudyProgrammeName());
+
+    UserEntityName loggedUserEntityName = userEntityController.getName(sessionController.getLoggedUserEntity(), false);
+    String loggedUserName = loggedUserEntityName.getDisplayName();
```

Comment:
Why move this to a separate variable? Places where this is used could simply say `loggedUserEntityName.getDisplayName()` anyway.